src/components/compress/index.tsx

```diff
@@ -470,14 +469,39 @@ export default class Compress extends Component<Props, State> {
 
     this.updateImageTimeoutIds[index] = self.setTimeout(
       () => {
-        this.updateImage(index, options).catch((err) => {
+        this.updateImage(index, options) .catch((err) => {
           console.error(err);
         });
       },
       delay,
     );
   }
 
+  @bind
+  private dispatchCustomEvent(name: string, detail?: {}) {
+    (this.base || document).dispatchEvent(new CustomEvent(name, { detail, bubbles: true }));
```

Comment:
Is there a benefit to firing it on `this.base`?

Reply:
I found that `this.base` is not set when we are just switching to from the `Intro` view to the `Compress` view. I couldn’t come up with a better workaround.

Reply:
Is there a disadvantage to just firing it on the document all the time?

Reply:
Not really no